lib/ansible/modules/cloud/amazon/aws_direct_connect_connection.py

```diff
@@ -0,0 +1,319 @@
+        if connection_name == each['connectionName'] and each['connectionState'] != 'deleted':
+            match.append(each['connectionId'])
+            connection.append(each)
+
+    # verifying if the connections exists; if true, return connection identifier, otherwise return False
+    if verify and len(match) == 1:
+        return match[0]
+    elif verify:
+        return False
+    # not verifying if the connection exists; just return current connection info
+    elif len(connection) == 1:
+        return {'connection': connection[0]}
+    return {'connection': {}}
+
+
+@AWSRetry.backoff(**retry_params)
+def create_connection(client, location, bandwidth, name, lag_id):
+    if not name:
+        raise DirectConnectError(msg="Failed to create a Direct Connect connection: name required.")
+    try:
+        if lag_id:
+            connection = client.create_connection(location=location,
+                                                  bandwidth=bandwidth,
+                                                  connectionName=name,
+                                                  lagId=lag_id)
+        else:
+            connection = client.create_connection(location=location,
+                                                  bandwidth=bandwidth,
+                                                  connectionName=name)
+    except botocore.exceptions.ClientError as e:
+        raise DirectConnectError(msg="Failed to create DirectConnect connection {0}".format(name),
+                                 last_traceback=traceback.format_exc(),
+                                 response=e.response)
+    return connection['connectionId']
+
+
+def changed_properties(current_status, location, bandwidth):
+    current_bandwidth = current_status['bandwidth']
+    current_location = current_status['location']
+
+    return (current_bandwidth != bandwidth or current_location != location)
+
+
+@AWSRetry.backoff(**retry_params)
+def update_associations(client, latest_state, connection_id, lag_id):
+    changed = False
+    if 'lagId' in latest_state and lag_id != latest_state['lagId']:
+        disassociate_connection_and_lag(client, connection_id, lag_id=latest_state['lagId'])
+        changed = True
+    if (changed and lag_id) or (lag_id and 'lagId' not in latest_state):
+        associate_connection_and_lag(client, connection_id, lag_id)
+        changed = True
+    return changed
+
+
+def ensure_present(client, connection_id, connection_name, location, bandwidth, lag_id, forced_update):
+    changed = False
+
+    # the connection is found; get the latest state and see if it needs to be updated
+    if connection_id:
+        latest_state = connection_status(client, connection_id=connection_id)['connection']
+        if changed_properties(latest_state, location, bandwidth) and forced_update:
+            ensure_absent(client, connection_id)
+            changed, connection_id = ensure_present(client=client,
```

Comment:
Instead, you can just return the results of `ensure_present` since this is the last line of this branch